share-through-htb.js

```diff
@@ -215,142 +183,79 @@ function ShareThroughHtb(configs) {
-         * returned demand.
-         *
-         * Use the adResponse variable to extract your bid information and insert it into the
-         * bids array. Each element in the bids array should represent a single bid and should
-         * match up to a single element from the returnParcel array.
-         *
-         */
-
-         /* ---------- Process adResponse and extract the bids into the bids array ------------*/
-
-        var bids = adResponse;
-
-        /* --------------------------------------------------------------------------------- */
-
-        for (var j = 0; j < returnParcels.length; j++) {
-            var curReturnParcel = returnParcels[j];
-
-            /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
-            var curBid;
-
-            for (var i = 0; i < bids.length; i++) {
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                 */
-
-                if (curReturnParcel.xSlotRef.someCriteria === bids[i].someCriteria) {
-                    curBid = bids[i];
-                    break;
-                }
-            }
-
-            /* ------------------------------------------------------------------------------------*/
-
-            /* HeaderStats information */
-            var headerStatsInfo = {};
-            headerStatsInfo[curReturnParcel.htSlot.getId()] = [curReturnParcel.xSlotName];
-
-            /* No matching bid found so its a pass */
-            if (!curBid) {
-                if (__profile.enabledAnalytics.requestTime) {
-                    __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
-                }
-                curReturnParcel.pass = true;
-                continue;
-            }
-
-            /* ---------- Fill the bid variables with data from the bid response here. ------------*/
-            /* Using the above variable, curBid, extract various information about the bid and assign it to
-            * these local variables */
-
-            var bidPrice = curBid.price; /* the bid price for the given slot */
-            var bidSize = [curBid.width, curBid.height]; /* the size of the given slot */
-            var bidCreative = curBid.adm; /* the creative/adm for the given slot that will be rendered if is the winner. */
-            var bidDealId = curBid.dealid; /* the dealId if applicable for this slot. */
-
-            /* ---------------------------------------------------------------------------------------*/
-
+        /* No matching bid found so its a pass */
+        if (curBid.creatives.length === 0) {
```

Comment:
Please add additional checks to ensure curBid is not undefined or evaluate to false.

Ex: `if (!curBid) {`

Reply:
Fixed. We now check for `(!curBid || !curBid.creatives || curBid.creatives.length === 0)`